### ISSUE TITLE
vue/html-closing-bracket-spacing rule is wrong

### ISSUE THREAD

**Issue by ffxsam** [vuejs/eslint-plugin-vue#909]
**Tell us about your environment**	
* **ESLint version:**	5.16.0
* **eslint-plugin-vue version:**	5.2.2
* **Node version:**	 10.16.0

**Please show your full configuration:**

```js
module.exports = {
  root: true,
  env: {
    node: true,
  },
  extends: ['plugin:vue/recommended', '@vue/standard', '@vue/typescript'],
  rules: {
    'array-callback-return': 'error',
    'array-element-newline': ['error', { multiline: true }],
    'arrow-spacing': ['error', { before: true, after: true }],
    camelcase: ['error', { properties: 'never' }],
    'comma-dangle': ['error', 'always-multiline'],
    curly: ['error', 'multi-line'],
    'default-case': 'error',
    'dot-location': ['error', 'property'],
    'dot-notation': 'error',
    eqeqeq: 'warn',
    'key-spacing': 'error',
    indent: ['error', 2, { SwitchCase: 1 }],
    'max-len': ['warn', { code: 80 }],
    'newline-per-chained-call': 'error',
    'no-alert': 'warn',
    'no-await-in-loop': 'error',
    'no-case-declarations': 'error',
    'no-console': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-const-assign': 'error',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-duplicate-imports': 'error',
    'no-else-return': 'error',
    'no-implicit-coercion': ['error', { allow: ['!!'] }],
    'no-mixed-spaces-and-tabs': 'error',
    'no-multi-spaces': 'error',
    'no-param-reassign': 'error',
    'no-return-await': 'error',
    'no-sequences': 'error',
    'no-trailing-spaces': 'error',
    'no-undef-init': 'error',
    'no-unreachable': 'warn',
    // 'no-unused-vars': 'warn',
    'no-useless-return': 'warn',
    'no-var': 'error',
    'object-curly-spacing': [
      'error',
      'always',
      { arraysInObjects: true, objectsInObjects: true },
    ],
    'padding-line-between-statements': [
      'error',
      { blankLine: 'any', prev: ['const', 'let'], next: ['const', 'let'] },
    ],
    'prefer-const': 'warn',
    'prefer-promise-reject-errors': 'warn',
    quotes: ['error', 'single'],
    semi: ['error', 'always'],
    'space-before-function-paren': [
      'error',
      {
        anonymous: 'never',
        named: 'never',
        asyncArrow: 'always',
      },
    ],
    'spaced-comment': ['error', 'always', {}],
    yoda: 'error',

    // Vue rules
    'vue/html-closing-bracket-newline': [
      'error',
      {
        singleline: 'never',
        multiline: 'always',
      },
    ],
  },
  parserOptions: {
    parser: '@typescript-eslint/parser',
  },
};
```

**What did you do?**	

```vue
<b-icon icon="align-left"/>
```

**What did you expect to happen?**
ESLint should not complain about this.

**What actually happened?**

```
warning: Expected a space before '/>', but not found (vue/html-closing-bracket-spacing) at src/components/ui/ButtonGroup.vue:9:34:
   7 |         size="is-small"
   8 |       >
>  9 |         <b-icon icon="align-left"/>
     |                                  ^
  10 |       </b-radio-button>
  11 |     </b-field>
  12 |   </div>
```

According to the [Vue Style Guide](https://vuejs.org/v2/style-guide/#Self-closing-components-strongly-recommended), `<MyComponent/>` with no space is recommended. Oddly, this ESLint rule is supposed to enforce that but does the opposite.

**Comment by ota-meshi** [comment]
Hello!

You can use the following settings to not allow spaces.

```json
  "vue/html-closing-bracket-spacing": ["error", {
    "startTag": "never",
    "endTag": "never",
    "selfClosingTag": "never"
  }]
```

**Comment by ffxsam** [comment]
@ota-meshi Thanks! I realize that, but I wanted to point out that by default, extending `plugin:vue/recommended` does not follow the official style guide. My point is that the default settings should raise an error or warning for `<MyComponent />`.

**Comment by ota-meshi** [comment]
Sorry for late reply.

I understand what you say. But changing the default setting is a breaking change, so I do not change it unless the style guide has documentation about the closing bracket spaces.

**Comment by kelvinmatheus** [comment]
> @ota-meshi Thanks! I realize that, but I wanted to point out that by default, extending `plugin:vue/recommended` does not follow the official style guide. My point is that the default settings should raise an error or warning for `<MyComponent />`.

I also prefer to have space before the `/>`. It is more readable. Our brain does not need to try to find where the name ends.

It also recommends using `kebab-case` when using the components. But there are many places that it uses `PascalCase`.

**Comment by ffxsam** [comment]
The convention I see the most is:

1. Use `kebab-case` for globally registered components.
2. Use `PascalCase` for components that are imported in the `<script>` block.

This is why third-party libraries like Vuetify use `v-progress-circular`, while your own components you create for your app and import will be `MyFunkyLoader`.

**Comment by ota-meshi** [comment]
I have no plans to change the default options, so I'll close this issue.
If you don't like the default options in the ruleset, you are free to override the configuration.

**Comment by ffxsam** [comment]
If you're worried about breaking changes, that's what major version releases are for.
